interview-questions.md

```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: It's a JavaScript built-in library that contains reusable components used for the front-end of a website. I'd use it because along with websites, it's used for mobile websites and applications.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: Constructor because it's the first phase out of the three phases and is the method that is always called when creating an object in a class.
 
-  Researched answer:
+  Researched answer: Rendor is a class method that produces an output on a website, it displays the specified HTML code. It's apart of the 'mounting' stage of react components lifecycle.   
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is an extension of JavaScript thus giving the ability to write HTML and the difference is that JSX uses 'nameClass' and HTML uses 'class'
 
-  Researched answer:
+  Researched answer: JSX stands JavaScript XML and is an extension HTML syntax. In react it converts the HTML tags into react components. It makes it easier to write code so instead of writing a code on seperate files we can write JavaScript along with HTML on one file and since class is already reserved className in camel case is used to prevent errors.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package manager that organizes the files attached from JEST
 
-  Researched answer:
+  Researched answer: Yarn was created by facebook as an open-source project and stands for 'Yet Another Resource Negotiator'. It manages and organizes files/packets for the node.js. 
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: Anonymous functions are functions that cannot be reassigned and are nameless. 
 
-  Researched answer:
+  Researched answer: When declared it isn't assigned an identifier so it can't be called upon when first created. It can be used as arguments for functions, the syntax are parenttheses which make it an expression that returns function objects. 
 
 
 ## Looking Ahead: Terms for Next Week
 
-1. Conditional rendering:
+1. Conditional rendering: Only rendoring certain components instead of the entire component.
 
-2. Object-oriented programming:
+2. Object-oriented programming: Also known as OOP is an alternative that allows for changes to be made to a specific part of the code versus making a change that affects the code entirety.
 
-3. Ruby:
+3. Ruby: A back-end programming language that is object-oriented and dynamically typed.
 
-4. Ruby blocks:
+4. Ruby blocks: Are anonymous functions that use the syntax do/end and can be passed to a method.
 
-5. Ruby hashes:
+5. Ruby hashes: Is a key value pair that stores data.
```

Comment:
Good job with the interview questions!  I really encourage you to expand on them, using examples or analogies :).